src/os/bluestore/BlueStore.h

```diff
@@ -1717,6 +1720,9 @@ class BlueStore : public ObjectStore,
 
   vector<Cache*> cache_shards;
 
+  std::mutex osr_lock;            ///< protect osd_set
+  std::set<OpSequencer*> osr_set; ///< set of all OpSequencers
```

Comment:
Replace with an intrusive list(or set if we need lookup support) for cheaper insert/removal? Will need more stuff in osr_drain_all to make a clone though...